### PR TITLE
Rescue Aws::EC2::Errors::InvalidSnapshotInUse

### PR DESCRIPTION
**Pull request by lastgabs**
### Description
When chef finds old snapshots, it tries to delete them. When the snapshot is being referenced in a AMI (Amazon Machine Image), it throws an Aws::EC2::Errors::InvalidSnapshotInUse and prevents chef from continuing running.

### Issues Resolved
No issues have been opened for this.

### Check List

- [ ] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [ ] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

**Comment by lastgabs** [comment]
I had to disable the `Lint/HandleExceptions` in order to make this work, but it's likely not ideal, open to suggestions on how else to deal with this.

File: resources/ebs_volume.rb [hunk 1]
```diff
@@ -131,7 +131,11 @@
     old_snapshots[new_resource.snapshots_to_keep, old_snapshots.length].each do |die|
       converge_by("delete snapshot with id: #{die[:snapshot_id]}") do
         Chef::Log.info "Deleting old snapshot #{die[:snapshot_id]}"
-        ec2.delete_snapshot(snapshot_id: die[:snapshot_id])
+        begin
+          ec2.delete_snapshot(snapshot_id: die[:snapshot_id])
+        rescue Aws::EC2::Errors::InvalidSnapshotInUse # rubocop: disable Lint/HandleExceptions
+          # Snapshot is being used by an AMI and therefore cannot be deleted.
```
Comment (tas50) [inline resources/ebs_volume.rb]: Can you add a Chef::Log.debug message here stating that the snapshot and ID cannot be removed

Comment (lastgabs) [reply resources/ebs_volume.rb]: Thanks for the review. Updated and added the message. Let me know if it's ok like this.

**Comment by tas50** [comment]
@lastgabs This looks good. Can you sign off the commits for the DCO so we can merge it in?

https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco

**Comment by lastgabs** [comment]
@tas50 thanks for the heads-up! The commit is now signed off

**Comment by tas50** [comment]
Thanks. I'll get this released

**Pull request merged**